spyder/plugins/projects/widgets/main_widget.py

```diff
@@ -598,6 +601,142 @@ def show_widget(self):
+            title = osp.basename(path)
+            icon = get_file_icon(path)
+            description = osp.dirname(path)
+            if len(path) > 75:
+                description = short_path
+            is_last_item = (i+1 == len(paths))
+
+            self.get_plugin()._switcher.add_item(
+                title=title,
+                description=description,
+                icon=icon,
+                section=section,
+                data=path,
+                last_item=is_last_item
+            )
+        self.get_plugin()._switcher.set_current_row(0)
+
+    def handle_switcher_selection(self, item, mode, search_text):
+        """
+        Handle user selecting item in switcher.
+        If the selected item is not in the section of the switcher that
+        corresponds to this plugin, then ignore it. Otherwise, switch to
+        selected item in notebook plugin and hide the switcher.
+        ----------
+        item: object
+            The current selected item from the switcher list (QStandardItem).
+        mode: str
+            The current selected mode (open files "", symbol "@" or line ":").
+        search_text: str
+            Cleaned search/filter text.
+        """
+
+        if item.get_section() != self.get_title():
+            return
+
+        # Open file in editor
+        filename = item.get_data()
+        switcher = self.get_plugin()._switcher
+        switcher.sig_open_file_requested.emit(filename)
+        switcher.hide()
+
+    def handle_switcher_results(self, search_text):
+        """
+        Handle user typing in switcher to filter result.
+        Load switcher results when a search text is typed for projects.
+        Parameters
+        ----------
+        text: str
+            The current search text in the switcher dialog box.
+        """
+        switcher = self.get_plugin()._switcher
+        paths = self._execute_fzf_subprocess(search_text)
+
+        for row in range(switcher.get_model().rowCount()):
+            item = switcher.get_model().item(row)
+            if isinstance(item, SwitcherItem):
+                if (item._data._filename.lower() in paths):
+                    paths.remove(item._data._filename.lower())
+
+        is_unsaved = [False] * len(paths)
+        short_paths = shorten_paths(paths, is_unsaved)
+        section = self.get_title()
+
+        for i, (path, short_path) in enumerate(zip(paths, short_paths)):
+            title = osp.basename(path)
+            icon = get_file_icon(path)
+            description = osp.dirname(path).lower()
+            if len(path) > 75:
+                description = short_path
+            is_last_item = (i+1 == len(paths))
+
+            switcher.add_item(
+                title=title,
+                description=description,
+                icon=icon,
+                section=section,
+                data=path,
+                last_item=is_last_item,
+                score=100
+            )
```

Comment:
Following the comments above, maybe the `add_item` call could go over the plugin's `handle_switcher_results` method side